arch/risc-v/src/cmake/Toolchain.cmake

```diff
@@ -343,21 +343,19 @@ if(CONFIG_RISCV_TOOLCHAIN STREQUAL GNU_RVG)
   # These models can't cover all implementation of RISCV, but it's enough for
   # most cases.
 
-  set(LLVM_CPUFLAGS)
-
   if(CONFIG_ARCH_RV32)
-    if(${ARCHCPUEXTFLAGS} STREQUAL imc)
-      list(APPEND LLVM_CPUFLAGS -mcpu=sifive-e20)
-    elseif(${ARCHCPUEXTFLAGS} STREQUAL imac)
-      list(APPEND LLVM_CPUFLAGS -mcpu=sifive-e31)
-    elseif(${ARCHCPUEXTFLAGS} STREQUAL imafc)
-      list(APPEND LLVM_CPUFLAGS -mcpu=sifive-e76)
+    if(${ARCHCPUEXTFLAGS} MATCHES "^imc")
+      set(LLVM_CPUTYPE "sifive-e20")
```

Comment:
but why use a specific vendor chip sifive-e20? @no1wudi

Reply:
For the RISC-V platform, the CPU_TYPE here is typically not utilized; it is merely used to denote a combination of ISAs. For instance, sifive-e20 represents IMC. This is because LLVM's ARCH_TYPE only specifies RISCV32/RISCV64 and does not reflect the specific ISA. In practical application scenarios, such as in the Rust compiler, sifive-e20 is decoded as riscv32imc and then passed to the Rust compiler.

Reply:
but why not pass the general riscv32imc directly?

Reply:
Because you can still pass it to compiler as `-mcpu` param in some case, it can works although it may not be the optimal choice, it will not cause any issues. But riscv32imc can not, it's not a valid identifier in LLVM backend:
https://github.com/apache/nuttx/blob/724797e05c8a90b9f9d170b021e403d88fc02dd3/tools/Zig.defs#L42-L46

For MCUs, symbols like sifive-e20 are merely used to indicate the ISA types supported by the RISC-V platform and do not entail any substantive optimizations for the CPU core itself.